### PR TITLE
CSIScaleOperator Status sub-resource should adhere to CloudPak Guidelines

### PR DESCRIPTION
**Pull request by nitishkumar4**
This pull request fixes the following issue:
- https://github.com/IBM/ibm-spectrum-scale-csi/issues/678

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [x] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

Status  sub-resource fields names and structure was not as per the guidelines.

## What is the new behavior?

- status subresource now adheres to CloudPak guidelines.
- updated status.conditions and status.versions fields
- updated status as per cloudpak guidelines.

```
Status:
  Conditions:
    Last Transition Time:  2022-07-19T04:49:20Z
    Message:               The CSI driver resources have been created/updated successfully.
    Reason:                successful
    Status:                True
    Type:                  Success
  Versions:
    Name:     ibm-spectrum-scale-csi
    Version:  2.7.0
```

```
# oc get cso
NAME                     VERSION   SUCCESS
ibm-spectrum-scale-csi   2.7.0     True
```

## How risky is this change?
- [x] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml [hunk 1]
```diff
@@ -1368,7 +1368,7 @@ spec:
             type: object
             description: CSIScaleOperatorStatus defines the observed state of CSIScaleOperator
             properties:
-              condition:
+              conditions:
```
[the remark this answers is not in the document]
Comment (nitishkumar4) [reply generated/installer/ibm-spectrum-scale-csi-operator-dev.yaml:1371]: did not observe any issue/error during upgrade.

**Comment by nitishkumar4** [comment]
Private Build: quay.io/nitish3356/csi:operatorStatus1

**Comment by saurabhwani5** [comment]
In upgrade , please show previous version instead of blanks , adding current logs below while upgrading
`[root@saurabh5-master ~]# oc get cso
NAME                     VERSION   SUCCESS
ibm-spectrum-scale-csi
[root@saurabh5-master ~]# oc get cso
NAME                     VERSION   SUCCESS
ibm-spectrum-scale-csi
[root@saurabh5-master ~]# oc get cso
NAME                     VERSION   SUCCESS
ibm-spectrum-scale-csi   2.7.0     True`

**Comment by nitishkumar4** [comment]
### Before
Status sub-resource fields:
- cso.status.condition
- cso.status.version

### After
- cso.status.conditions
- cso.status.versions

Since the field paths are updated, anyone relying on older paths will have to update the path.

**Review by amdabhad — APPROVED** [review]

**Review by Jainbrt — APPROVED** [review]

**Pull request merged**